network/src/protocols/identify/mod.rs

```diff
@@ -457,8 +457,15 @@ impl Callback for IdentifyCallback {
                     // and if set it to peer store, it will be broadcast to the entire network,
                     // but this is an unverified address
                     if renew {
+                        let dns_addr = self.network_state.pending_dns_addrs.write().remove(
```

Comment:
A address will be written to the `pending_dns_addrs` field when the session open, but it is only removed when the identity protocol succeeds, if the identity protocol fails, then this field will keep growing, it is recommended to remove it when the session is closed.